app/client/jsx/ImageMapContents.jsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react'
+import { PDFObject } from 'react-pdfobject'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faTimes, faArrowLeft } from '@fortawesome/free-solid-svg-icons'
+
+function getContentsHtml(contentsList, onClickFunc) {
+    return contentsList.map((c,i) => 
+        <div className="item" key={i} onClick={() => onClickFunc(c)}>{c.title}</div>
+    )
+}
+
+function displayList(props, onClickFunc) {
+    return (
+        <div className="imagemap-modal">
+            <div className="header">
+                <b>{props.area.label}</b>
+                <button onClick={props.handleClose}>
+                    <FontAwesomeIcon icon={faTimes}/>
+                </button>
+            </div>
+            <div className="contents">
+                {getContentsHtml(props.area.contents, onClickFunc)}
+            </div>
+        </div>
+    )
+}
+
+function onSelectItem(setItemIsShown, setItem) {
+    return item => {
+        setItem(item)
+        setItemIsShown(true)
+    }
+}
+
+function displayItem(props, item, returnFunc) {
+    return (
+        <div className="imagemap-modal">
+            <div className="header">
+                <button onClick={returnFunc}>
+                    <FontAwesomeIcon icon={faArrowLeft}/>
+                </button>
+                <b>{item.title}</b>
+                <button onClick={props.handleClose}>
+                    <FontAwesomeIcon icon={faTimes}/>
+                </button>
+            </div>
+            <div className="contents">
+                <PDFObject url={item.path} />
```

Comment:
Does this pdf component allow us to show images as well? I'm not really sure how PDF viewers work

Reply:
I assume you mean, allow `contents` elements that are .jpg or .png in addition to .pdf? (As opposed to showing a preview image of a PDF as mentioned above).

I don't think it supports images natively, but we could add logic around the path to detect the filetype and display it in different ways.